### ISSUE TITLE
Make all our major classes (Wallet, ChainHandler, KeyManager etc) have options to do `.fromDefaultDatadir()`

### ISSUE THREAD

**Issue by Christewart** [bitcoin-s/bitcoin-s#2563]
Right now we have to do all this manual configuration in docs 

https://bitcoin-s.org/docs/next/wallet/wallet#example

I don't think there is a reason we can't just make a super factory trait that has `.fromDefaultDatadir()` similar to what we have in [`AppConfigFactory.fromDefaultDatadir()`](https://github.com/bitcoin-s/bitcoin-s/blob/ee7c96245e99325d1386464c4462d36dce7bcafb/db-commons/src/main/scala/org/bitcoins/db/AppConfigFactory.scala#L9)

I think this will simplify things greatly.

**Comment by benthecarman** [comment]
I don't think this is really possible, for example the wallet needs a `keyManager`, `nodeApi`, `chainApi`, `feeRateApi`, and a `creationTime` and all of these have different params that we need to take in.

**Comment by Christewart** [comment]
Hmm, i think since the [`wallet.dependsOn(keyManager)`](https://github.com/bitcoin-s/bitcoin-s/blob/b0b56dd5da2f7e3c2428008d9a3c9f7391cf53e1/build.sbt#L538) we could do that project, but you do have a point with the `nodeApi`/`chainApi` as that isn't a dependency of the wallet :/.
